x-pack/legacy/plugins/siem/public/containers/case/api.test.tsx

```diff
@@ -0,0 +1,402 @@
+    });
+
+    test('check url, method, signal', async () => {
+      await patchComment(
+        basicCase.id,
+        basicCase.comments[0].id,
+        'updated comment',
+        basicCase.comments[0].version,
+        abortCtrl.signal
+      );
+      expect(fetchMock).toHaveBeenCalledWith(`${CASES_URL}/${basicCase.id}/comments`, {
+        method: 'PATCH',
+        body: JSON.stringify({
+          comment: 'updated comment',
+          id: basicCase.comments[0].id,
+          version: basicCase.comments[0].version,
+        }),
+        signal: abortCtrl.signal,
+      });
+    });
+
+    test('happy path', async () => {
+      const resp = await patchComment(
+        basicCase.id,
+        basicCase.comments[0].id,
+        'updated comment',
+        basicCase.comments[0].version,
+        abortCtrl.signal
+      );
+      expect(resp).toEqual(basicCase);
+    });
+  });
+  describe('postCase', () => {
+    beforeEach(() => {
+      fetchMock.mockClear();
+      fetchMock.mockResolvedValue(basicCaseSnake);
+    });
+    const data = {
+      description: 'description',
+      tags: ['tag'],
+      title: 'title',
+    };
+
+    test('check url, method, signal', async () => {
+      await postCase(data, abortCtrl.signal);
+      expect(fetchMock).toHaveBeenCalledWith(`${CASES_URL}`, {
+        method: 'POST',
+        body: JSON.stringify(data),
+        signal: abortCtrl.signal,
+      });
+    });
+
+    test('happy path', async () => {
+      const resp = await postCase(data, abortCtrl.signal);
+      expect(resp).toEqual(basicCase);
+    });
+  });
+  describe('postComment', () => {
+    beforeEach(() => {
+      fetchMock.mockClear();
+      fetchMock.mockResolvedValue(basicCaseSnake);
+    });
+    const data = {
+      comment: 'comment',
+    };
+
+    test('check url, method, signal', async () => {
+      await postComment(data, basicCase.id, abortCtrl.signal);
+      expect(fetchMock).toHaveBeenCalledWith(`${CASES_URL}/${basicCase.id}/comments`, {
+        method: 'POST',
+        body: JSON.stringify(data),
+        signal: abortCtrl.signal,
+      });
+    });
+
+    test('happy path', async () => {
+      const resp = await postComment(data, basicCase.id, abortCtrl.signal);
+      expect(resp).toEqual(basicCase);
+    });
+  });
+  describe('pushCase', () => {
+    beforeEach(() => {
+      fetchMock.mockClear();
+      fetchMock.mockResolvedValue(pushedCaseSnake);
+    });
+
+    test('check url, method, signal', async () => {
+      await pushCase(basicCase.id, pushSnake, abortCtrl.signal);
+      expect(fetchMock).toHaveBeenCalledWith(`${CASES_URL}/${basicCase.id}/_push`, {
+        method: 'POST',
+        body: JSON.stringify(pushSnake),
+        signal: abortCtrl.signal,
+      });
+    });
+
+    test('happy path', async () => {
+      const resp = await pushCase(basicCase.id, pushSnake, abortCtrl.signal);
+      expect(resp).toEqual(pushedCase);
+    });
+  });
+  describe('pushToService', () => {
```

Comment:
I think we should create a test that checks if the function throws an error when the `response.status` is `error`.